tasks/diff.py

```diff
@@ -0,0 +1,189 @@
+        "agent": {
+            "entrypoint": "cmd/agent",
+            "platforms": ["linux/x64", "linux/arm64", "win32/x64", "win32/x86", "darwin/x64", "darwin/arm64"],
+        },
+        "iot-agent": {
+            "build": "agent",
+            "entrypoint": "cmd/agent",
+            "flavor": AgentFlavor.iot,
+            "platforms": ["linux/x64", "linux/arm64"],
+        },
+        "heroku-agent": {
+            "build": "agent",
+            "entrypoint": "cmd/agent",
+            "flavor": AgentFlavor.heroku,
+            "platforms": ["linux/x64"],
+        },
+        "cluster-agent": {"entrypoint": "cmd/cluster-agent", "platforms": ["linux/x64", "linux/arm64"]},
+        "cluster-agent-cloudfoundry": {
+            "entrypoint": "cmd/cluster-agent-cloudfoundry",
+            "platforms": ["linux/x64", "linux/arm64"],
+        },
+        "dogstatsd": {"entrypoint": "cmd/dogstatsd", "platforms": ["linux/x64", "linux/arm64"]},
+        "process-agent": {
+            "entrypoint": "cmd/process-agent",
+            "platforms": ["linux/x64", "linux/arm64", "win32/x64", "darwin/x64", "darwin/arm64"],
+        },
+        "heroku-process-agent": {
+            "build": "process-agent",
+            "entrypoint": "cmd/process-agent",
+            "flavor": AgentFlavor.heroku,
+            "platforms": ["linux/x64"],
+        },
+        "security-agent": {
+            "entrypoint": "cmd/security-agent",
+            "platforms": ["linux/x64", "linux/arm64"],
+        },
+        "serverless": {"entrypoint": "cmd/serverless", "platforms": ["linux/x64", "linux/arm64"]},
+        "system-probe": {"entrypoint": "cmd/system-probe", "platforms": ["linux/x64", "linux/arm64", "win32/x64"]},
+        "trace-agent": {
+            "entrypoint": "cmd/trace-agent",
+            "platforms": ["linux/x64", "linux/arm64", "win32/x64", "win32/x86", "darwin/x64", "darwin/arm64"],
+        },
+        "heroku-trace-agent": {
+            "build": "trace-agent",
+            "entrypoint": "cmd/trace-agent",
+            "flavor": AgentFlavor.heroku,
+            "platforms": ["linux/x64"],
+        },
+    }
+    diffs = {}
+    dep_cmd = "go list -f '{{ range .Deps }}{{ printf \"%s\\n\" . }}{{end}}'"
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        try:
+            # generate list of imports for each target+branch combo
+            branches = {"current": None, "main": baseline_ref}
+            for branch_name, branch_ref in branches.items():
+                if branch_ref:
+                    ctx.run(f"git checkout -q {branch_ref}")
+
+                for binary, details in binaries.items():
+                    with ctx.cd(details.get("entrypoint")):
+                        for combo in details.get("platforms"):
+                            platform, arch = combo.split("/")
+                            goos, goarch = GOOS_MAPPING.get(platform), GOARCH_MAPPING.get(arch)
+                            target = f"{binary}-{goos}-{goarch}"
```

Comment:
Should the target also contain the flavor? If it doesn't, I think that e.g. the heroku process agent and regular process agent will try to use the same file and will end up overwriting it, won't they?

Reply:
The flavor is already built into the `binary` variable, like `heroku-agent` and `heroku-process-agent`.